rosbag2_transport/test/rosbag2_transport/test_record_all_use_sim_time.cpp

```diff
@@ -0,0 +1,91 @@
+// Copyright 2018, Bosch Software Innovations GmbH.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gmock/gmock.h>
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "rclcpp/rclcpp.hpp"
+
+#include "rosbag2_test_common/publication_manager.hpp"
+#include "rosbag2_test_common/wait_for.hpp"
+
+#include "rosbag2_transport/recorder.hpp"
+
+#include "test_msgs/msg/basic_types.hpp"
+#include "test_msgs/message_fixtures.hpp"
+
+#include "record_integration_fixture.hpp"
+#include "rosgraph_msgs/msg/clock.hpp"
+
+std::vector<rosgraph_msgs::msg::Clock::SharedPtr> get_clock_messages()
+{
+  std::vector<rosgraph_msgs::msg::Clock::SharedPtr> messages;
+  for (int i = 0; i < 5; ++i) {
+    auto msg = std::make_shared<rosgraph_msgs::msg::Clock>();
+    msg->clock.sec = 1234567890;
+    msg->clock.nanosec = 123456789;
+    messages.push_back(msg);
+  }
+  return messages;
+}
+
+TEST_F(RecordIntegrationTestFixture, record_all_with_sim_time)
+{
+  std::string string_topic = "/string_topic";
+  auto string_message = get_messages_strings()[0];
+  string_message->string_value = "Hello World";
+
+  // clock
+  std::string clock_topic = "/clock";
+  auto clock_message = get_clock_messages()[0];
+
+  rosbag2_test_common::PublicationManager pub_manager;
+  pub_manager.setup_publisher(string_topic, string_message, 5);
+  pub_manager.setup_publisher(clock_topic, clock_message, 5);
+
+  rosbag2_transport::RecordOptions record_options = {true, true, {}, "rmw_format", 100ms};
```

Comment:
Ok. I've figured out the test works because we do initial discovery on start even if discovery disabled.

Nevertheless still highly recommend to provide list of topics explicitly to avoid interference with unexpected messages.

Reply:
Thanks! I didn't know that I needed to add a topic list to RecorderOption.
4d37642

Reply:
@KeisukeShima If you will not specify topic list it will record all available topics. Including `/rosout` which will interfere with tests expectations. 
Thanks for adding list of topics.